File: .sdk-repos/orchestration-cluster-api-python/examples/deployment.py
# Compilable usage examples for deployment operations.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import CamundaClient


# region DeployResources
def deploy_resources_example() -> None:
    client = CamundaClient()

    result = client.deploy_resources_from_files(
        ["order-process.bpmn", "decision.dmn"]
    )

    print(f"Deployment key: {result.deployment_key}")
    for process in result.processes:
        print(
            f"  Process: {process.process_definition_id} v{process.process_definition_version}"
        )
    for decision in result.decisions:
        print(f"  Decision: {decision.decision_definition_id}")
# endregion DeployResources


# region DeployResourcesWithTenant
def deploy_resources_with_tenant_example() -> None:
    client = CamundaClient()

    result = client.deploy_resources_from_files(
        ["order-process.bpmn"],
        tenant_id="my-tenant",
    )

    print(f"Deployment key: {result.deployment_key}")
    print(f"Tenant: {result.tenant_id}")
# endregion DeployResourcesWithTenant


# region DeleteResource
def delete_resource_example() -> None:
    client = CamundaClient()

    # Use a resource key from a previous deployment response
    client.delete_resource(resource_key="2251799813685249")
# endregion DeleteResource

File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/Deployment.cs
// Compilable usage examples for deployment and resource operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class DeploymentExamples
{
    #region CreateDeployment
    // <CreateDeployment>
    public static async Task CreateDeploymentExample()
    {
        using var client = CamundaClient.Create();

        var content = new MultipartFormDataContent();
        var fileContent = new ByteArrayContent(File.ReadAllBytes("process.bpmn"));
        content.Add(fileContent, "resources", "process.bpmn");

        var result = await client.CreateDeploymentAsync(content);
        Console.WriteLine($"Deployment key: {result.DeploymentKey}");
    }
    // </CreateDeployment>
    #endregion CreateDeployment

    #region DeleteResource

    // <DeleteResource>
    public static async Task DeleteResourceExample(ResourceKey resourceKey)
    {
        using var client = CamundaClient.Create();

        await client.DeleteResourceAsync(
            resourceKey,
            new DeleteResourceRequest());
    }
    // </DeleteResource>
    #endregion DeleteResource

    #region GetResource

    // <GetResource>
    public static async Task GetResourceExample(ResourceKey resourceKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetResourceAsync(resourceKey);
        Console.WriteLine($"Resource: {result.ResourceName}");
    }
    // </GetResource>
    #endregion GetResource

    #region GetResourceContent

    // <GetResourceContent>
    public static async Task GetResourceContentExample(ResourceKey resourceKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetResourceContentAsync(resourceKey);
        Console.WriteLine($"Content: {result}");
    }
    // </GetResourceContent>
    #endregion GetResourceContent

    #region DeployResourcesFromFiles

    // <DeployResourcesFromFiles>
    public static async Task DeployResourcesFromFilesExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.DeployResourcesFromFilesAsync(
            ["process.bpmn", "decision.dmn"]);
        Console.WriteLine($"Deployment key: {result.DeploymentKey}");
    }
    // </DeployResourcesFromFiles>
    #endregion DeployResourcesFromFiles
}
